prov/sm2/src/sm2.h

```diff
@@ -279,13 +282,11 @@ struct sm2_ep {
 	size_t rx_size;
 	const char *name;
 	uint64_t msg_id;
-	struct sm2_region *volatile region;
-	// if double locking is needed, shm region lock must
-	// be acquired before any shm EP locks
+	int self_fiaddr;
+	struct sm2_mmap *mmap_regions;
```

Comment:
It is easy to get from the ep to the map since it is part of the sm2_av (util_ep->av). No need to double this pointer.

Reply:
fixed